### PR TITLE
fix: resolve problem with severity change for configurable rules

### PR DESCRIPTION
**Pull request by RomanHotsiy**
## What/Why/How?

Severity of configurable rules could not be overridden without full rule copy:

**base.yaml**
```yaml
rules:
  rule/abc:
    severity: error
    # ...
```

**redocly.yaml**
```yaml
extends:
  - base.yaml
rules:
  rule/abc: warn
```

The above did not work.

## Reference
Reported by a customer.

Related to https://github.com/Redocly/redocly/issues/12111

## Testing

Covered with unit test.

## Check yourself

- [ ] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests
- [ ] New package installed? - Tested in different environments (browser/node)

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/config/utils.ts [hunk 1]
```diff
@@ -198,47 +199,47 @@ export function mergeExtends(rulesConfList: ResolvedStyleguideConfig[]) {
       );
     }
 
-    Object.assign(result.rules, rulesConf.rules);
-    Object.assign(result.oas2Rules, rulesConf.oas2Rules);
-    assignExisting(result.oas2Rules, rulesConf.rules || {});
```
Comment (tatomyr) [inline packages/core/src/config/utils.ts:203]: Why removed `|| {}`? Either remove it for all cases or add it back here. Actually, I see you've added a [check](https://github.com/Redocly/redocly-cli/pull/1811/files#diff-6552eacdc0f4378c00e24a96877df8352eeb0db006852dfdcf511f59d9b781aeR217) for an empty object, so the fallback could be removed.

File: packages/core/src/utils.ts [hunk 2]
```diff
@@ -196,9 +196,30 @@ export function isNotString<T>(value: string | T): value is T {
...
+};
+
+export function assignExistingRules<T extends string | { severity?: string }>(
+  target: Record<string, T>,
+  obj?: Record<string, T>
+) {
+  if (!obj) return;
+  for (const k of Object.keys(obj)) {
+    if (!target.hasOwnProperty(k)) continue;
+    if (typeof target[k] === 'object' && target[k] !== null && typeof obj[k] === 'string') {
```
Comment (tatomyr) [inline packages/core/src/utils.ts:220]: Please use `isPlainObject`.

File: packages/core/src/utils.ts [hunk 3]
```diff
@@ -196,9 +196,30 @@ export function isNotString<T>(value: string | T): value is T {
   return !isString(value);
 }
 
-export function assignExisting<T>(target: Record<string, T>, obj: Record<string, T>) {
+export const assignRules = <T extends string | { severity?: string }>(
```
Comment (tatomyr) [inline packages/core/src/utils.ts:199]: I see it's being applied to decorators and preprocessors as well. Could you come up with a more generic/descriptive name?

File: .changeset/fuzzy-crews-boil.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/openapi-core": patch
+"@redocly/cli": patch
+---
+
+Resolved an issue where overrides for the severity of configurable scorecard rules were ignored.
```
Comment (tatomyr) [inline .changeset/fuzzy-crews-boil.md:6]: Looks like it's not just about scorecard rules but configurable rules in general.

**Review by tatomyr — COMMENTED** [review]
Left a few comments.

**Review by JLekawa — APPROVED** [review]

File: .changeset/fuzzy-crews-boil.md (hunk 4 above)
Comment (JLekawa) [inline .changeset/fuzzy-crews-boil.md:6]:
```suggestion
Resolved an issue where overrides for the severity of configurable rules were ignored.
```

File: packages/core/src/config/utils.ts (hunk 1 above)
Comment (RomanHotsiy) [reply packages/core/src/config/utils.ts:203]: oh, had to be for all cases, thanks. removed everywhere

**Review by tatomyr — APPROVED** [review]
LGMT, thanks!

**Pull request merged**